release/models/nat/openconfig-nat.yang

```diff
@@ -0,0 +1,802 @@
+      description
+        "Total number of NAT mappings created since instance startup";
+    }
+
+    leaf mapping-failures {
+      type yang:counter64;
+      description
+        "Number of mapping creation failures";
+    }
+  }
+
+  grouping nat-interface-config {
+    description
+      "Configuration parameters for NAT interfaces";
+
+    leaf interface {
+      type oc-if:base-interface-ref;
+      description
+        "Reference to the interface";
+    }
+
+    leaf direction {
+      type nat-direction;
+      description
+        "NAT translation direction for this interface";
+    }
+  }
+
+  grouping nat-interface-state {
+    description
+      "Operational state data for NAT interfaces";
+
+    leaf packets-translated {
+      type yang:counter64;
+      description
+        "Number of packets that have been translated";
+    }
+
+    leaf bytes-translated {
+      type yang:counter64;
+      description
+        "Number of bytes that have been translated";
+    }
+
+    leaf translation-errors {
+      type yang:counter64;
+      description
+        "Number of translation errors encountered";
+    }
+  }
+
+  grouping nat-interface-top {
+    description
+      "Top-level grouping for NAT interface data";
+
+    container interfaces {
+      description
+        "Interface configuration and state for NAT";
+
+      list interface {
+        key "interface";
+        description
+          "List of interfaces participating in NAT";
+
+        leaf interface {
+          type leafref {
+            path "../config/interface";
+          }
+          description
+            "Reference to interface list key";
+        }
+
+        container config {
+          description
+            "Configuration parameters for the NAT interface";
+
+          uses nat-interface-config;
+        }
+
+        container state {
+          config false;
+          description
+            "Operational state data for the NAT interface";
+
+          uses nat-interface-config;
+          uses nat-interface-state;
+        }
+      }
+    }
+  }
+
+  grouping nat-pool-config {
+    description
+      "Configuration parameters for NAT address pools";
+
+    leaf name {
+      type string;
+      description
+        "Unique name for the NAT pool";
+    }
+
+    leaf start-address {
+      type inet:ipv4-address;
+      description
+        "Starting IPv4 address of the pool range";
+    }
+
+    leaf end-address {
+      type inet:ipv4-address;
+      description
+        "Ending IPv4 address of the pool range";
+    }
+
+    leaf port-range-start {
+      type inet:port-number;
+      description
+        "Starting port number for NAPT translations";
+    }
+
+    leaf port-range-end {
+      type inet:port-number;
+      description
+        "Ending port number for NAPT translations";
+    }
+
+    leaf port-block-size {
+      type uint16;
+      description
+        "Number of ports allocated per mapping";
+    }
+
+    choice traffic-matcher {
```

Comment:
[OC style](https://github.com/openconfig/public/blob/692101403054d60763cc252df54ad0812e250169/doc/openconfig_style_guide.md#choice) strongly prefers to avoid choice statements, but you could use when for enforcing some validations in the model.  (So far we have not used `choice` in any OC model)

Reply:
Thanks @dplore, that was an oversight on our end. We replaced `choice` with `when` statements.